common/source/serial_port.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * serial_port.cpp
+ *
+ *  Created on: 2020��9��18��
+ *      Author: duanning
+ */
+
+#include "serial_port.h"
+
+SerialPort::SerialPort(const string& name):
+	serial_port_(io_s_) {
+	CommonKeyPairs config = {
+			{"hid_name", name},
+	};
+
+	ReConfig(config);
+}
+
+SerialPort::SerialPort(const string& name, AbstractProtocol& protocol):
+	serial_port_(io_s_),
+	application_protocol_(protocol) {
+	CommonKeyPairs config = {
+			{"hid_name", name},
+	};
+
+	ReConfig(config);
+}
+
+SerialPort::SerialPort(string name, int baud_rate, character_size_type character_size, \
+		parity_type parity, stop_bits_type stop_bits, flow_control_type flow_control):
+		serial_port_(io_s_) {
+	CommonKeyPairs config = {
+			{"serial_port_name", name},
+	};
+
+	serial_params.kByteCommBaudRate = baud_rate;
+	serial_params.kByteCommCharacterSize = character_size;
+	serial_params.kByteCommParity = parity;
+	serial_params.kByteCommStopBits = stop_bits;
+	serial_params.kByteCommFlowControl = flow_control;
+
+	if (ReConfig(config) != kNoError)
+		return;
+}
+
+SerialPort::~SerialPort() {
+
+}
+
+error_condition SerialPort::ReConfig(const CommonKeyPairs &config) {
+	ClearInitialized();
+	auto serial_port_name = config.find(kSerialPortName);
```

Comment:
As I sugguested, to use string key name, we need to also change this function. When calling `Set` function, we could append a key pair to a private config member.

Reply:
Please store the config in map.

Reply:
I have already did it